docs/example.rst

```diff
@@ -0,0 +1,269 @@
+Example: A small tuning example using sparse matrices
+=====================================================
+
+:code:`tuneit` is imported as shown below along with some other packages that are used in this example:
+
+.. code-block:: python
+
+   from tuneit import *
+   import scipy.sparse as sp
+   import numpy as np
+
+Firstly, a simple graph is constructed, which computes the multiplication of a sparse matrix with a vector. 
+The graph contains one variable to be tuned, which represents the different formats that can be used for the sparse matrix.
+The following function creates the sparse matrix and by using :code:`alternatives` more options are added for the format that will 
+be used to express the matrix (available in the :code:`scipy.sparse` package).
+
+.. code-block:: python
+
+   @alternatives(
+       var_name='which_sparse',
+       coo = lambda matrix: sp.coo_matrix(matrix),
+       csc = lambda matrix: sp.csc_matrix(matrix),
+       csr = lambda matrix: sp.csr_matrix(matrix)
+   )
+   def bsr(matrix): 
+       return sp.bsr_matrix(matrix)
+
+In this way, we have created a function :code:`bsr` that expresses the given sparse matrix in an appropriate format and a variable 
+to be tuned. The range of the variable :code:`which_sparse` contains all different options that can be used to express the sparse matrix, which are included in the function above (:code:`coo,csc,csr,bsr`). 
+
+The graph takes as input the matrix and the vector to be multiplied. One option is to create a matrix and a vector at random:
+
+.. code-block:: python
+
+   mat=scipy.sparse.random(100,100,0.1)
+   vec=np.random.rand(100)
+
+Or just create two generic data objects, which will take their actual value later on: (this is the option used in this example)
+
+.. code-block:: python
+
+   mat=data(info=["shape","dtype"])
+   vec=data(info=["shape","dtype"])
+  
+:code:`data()` creates a generic data object as no specific value is passed in the function. Even though no value is passed, some information 
+about the new data object can be given using :code:`info`. As shown above, some characteristics about the new objects are given by the 
+attributes :code:`shape` and :code:`dtype`. 
+
+In addition, the :code:`bsr` function constructed previously can now be used. The new :code:`mat` object created after 
+:code:`bsr` is called on the object :code:`mat` (created above) is a tunable object.
+
+.. code-block:: python
+
+   mat=bsr(mat)
+
+The final graph :code:`mul` that expresses the multiplication between the vector :code:`vec` and the sparse matrix :code:`mat` is created 
+as shown below:
+
+.. code-block:: python
+
+   mul=finalize(mat*vec)
+
+.. 
+   Do I need more explanations about the finalize function here and why it is needed?
```

Comment:
Yes, in the above snippet just write `mul = mat * vec`.
Then say something like:
Once the graph is completed we can finalize it with the function `finalize`.
```
mul = finalize(mul)
```
This closes the graph and provides us an high-level interface for processing the graph. 
E.g. we can simply compute it by calling it
```
mat = sp.random(100,100,0.1)
￼vec = np.random.rand(100)

out = mul(mat=mat, vec=vec)
```